# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchService.h
# Commit: cf972fe0d961
# Full Hash: cf972fe0d96164788f998f04008930474526e400
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r f8afd25bf41e -r cf972fe0d961 dom/fetch/FetchService.h
--- a/dom/fetch/FetchService.h	Sat Feb 26 08:40:25 2022 +0000
+++ b/dom/fetch/FetchService.h	Sat Feb 26 08:40:26 2022 +0000
@@ -1,5 +1,3 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
@@ -12,6 +10,7 @@
 #include "mozilla/MozPromise.h"
 #include "mozilla/RefPtr.h"
 #include "mozilla/dom/FetchDriver.h"
+#include "mozilla/dom/PerformanceTimingTypes.h"
 #include "mozilla/dom/SafeRefPtr.h"
 
 class nsILoadGroup;
@@ -24,8 +23,12 @@
 class InternalRequest;
 class InternalResponse;
 
+using FetchServiceResponse =
+    Tuple<SafeRefPtr<InternalResponse>, IPCPerformanceTimingData, nsString,
+          nsString>;
+
 using FetchServiceResponsePromise =
-    MozPromise<SafeRefPtr<InternalResponse>, CopyableErrorResult, true>;
+    MozPromise<FetchServiceResponse, CopyableErrorResult, true>;
 
 /**
  * FetchService is a singleton object which designed to be used in parent